src/vs/platform/extensionManagement/node/extensionGalleryService.ts

```diff
@@ -466,6 +469,43 @@ export class ExtensionGalleryService implements IExtensionGalleryService {
 			if (names && names.length > 0) {
 				filteredExtensions = filteredExtensions.filter(e => e.extensionName && e.publisher.publisherName && names.includes(`${e.publisher.publisherName.toLocaleLowerCase()}.${e.extensionName.toLocaleLowerCase()}`));
 			}
+			const searchTexts = query.criteria.filter(x => x.filterType === FilterType.SearchText).map(v => v.value.toLocaleLowerCase());
+			if (searchTexts && searchTexts.length > 0) {
+				searchTexts.forEach(searchText => {
+					if (searchText !== '@allmarketplace') {
+						filteredExtensions = filteredExtensions.filter(
+							e => 	e.extensionName && e.extensionName.includes(searchText) ||
+									e.publisher && e.publisher.publisherName && e.publisher.publisherName.includes(searchText) ||
+									e.publisher && e.publisher.displayName && e.publisher.displayName.includes(searchText) ||
+									e.displayName && e.displayName.includes(searchText) ||
+									e.shortDescription && e.shortDescription.includes(searchText) ||
+									e.extensionId && e.extensionId.includes(searchText)
+						);
+					}
+				});
+			}
+		}
+
+		// Sorting
+		switch (query.sortBy) {
+			case SortBy.PublisherName:
+				filteredExtensions.sort((a: IRawGalleryExtension, b: IRawGalleryExtension) => {
+					if (a && b && a.publisher && a.publisher.publisherName && b.publisher && b.publisher.publisherName) {
+						return a.publisher.publisherName < b.publisher.publisherName ? -1 : 1;
+					}
+					return 0;
+				});
+				break;
+			case SortBy.Title:
+				filteredExtensions.sort((a: IRawGalleryExtension, b: IRawGalleryExtension) => {
+					if (a && b && a.displayName && b.displayName) {
+						return a.displayName < b.displayName ? -1 : 1;
+					}
+					return 0;
```

Comment:
seems like this will have arbitrary sort order if one extension has no display name and the other does.  Not sure it's a problem, just seems non-deterministic